test/WebJobs.Script.Tests.Integration/WebHostEndToEnd/SamplesEndToEndTests_CSharp.cs

```diff
@@ -212,11 +219,19 @@ public async Task SetHostState_Offline_Succeeds()
             // wait for the host to go offline
             await AwaitHostStateAsync(ScriptHostState.Offline);
 
+            // verify function status returns 503 immediately
+            await TestHelpers.RunWithTimeoutAsync(async () =>
```

Comment:
This test would hang for a minute w/o the fix

Reply:
👍